### ISSUE TITLE
Warning messages are printed with colors that are invisible on white background

### ISSUE THREAD

**Issue by brianhuffman** [commercialhaskell/stack#4047]
Here's an example of a warning message that I received from a recent run of `stack build`:
```
Warning: addDependentFile path (Template Haskell) listed in
         /Users/huffman/Work/saw-script/deps/cryptol/.stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/src/Cryptol/Prelude.dump-hi
         does not exist: lib/Cryptol/Extras.cry
```
The problem is that on my terminal, which uses the default black-text-on-white-background theme, it looks like this:

![stack-warning](https://user-images.githubusercontent.com/619882/40734732-230e93ea-63ee-11e8-81a4-fe8f52e7b618.png)

The filenames are printed in a completely invisible bold white, and the word "Warning" is printed in a yellow color that is just barely visible against a white background. The text is only readable after highlighting it with the mouse.

To reproduce: Using a terminal with a dark-text-on-light-background theme, run any `stack` command that prints any kind of warning message. 

Here is the output of `stack --version`:
```
Version 1.7.1, Git revision 681c800873816c022739ca7ed14755e85a579565 (5807 commits) x86_64 hpack-0.28.2
```

**Comment by brianhuffman** [comment]
The offending lines in the source code are https://github.com/commercialhaskell/stack/blob/master/src/Stack/PrettyPrint.hs#L167
```
styleWarning :: AnsiDoc -> AnsiDoc
styleWarning = yellow
```
and https://github.com/commercialhaskell/stack/blob/master/src/Stack/PrettyPrint.hs#L181
```
styleFile :: AnsiDoc -> AnsiDoc
styleFile = bold . white
```

**Comment by mattaudesse** [comment]
Thanks @brianhuffman - that's definitely aggravating.

It's not a full solution, but are you aware of the `--color never` option?

It looks like you're using macOS. Are you running the standard terminal or something else (e.g. `iterm2`)?

**Comment by brianhuffman** [comment]
Yes, I'm on MacOS, and I usually use iterm2.

**Comment by mgsloan** [comment]
The actual colors chosen are determined by your terminal, so I do not think that this is something that needs to be addressed in stack.

**Comment by brianhuffman** [comment]
@mgsloan Are you suggesting that I should remap the color "white" in my terminal to something dark?

**Comment by mgsloan** [comment]
Or remap the yellow to a darker shade.  CLI apps are going to be using these colors, so they should be configured to be readable.

**Comment by brianhuffman** [comment]
I can deal with the yellow. The white-on-white is more of a problem. iTerm2 is configurable enough so that there are some workarounds for it (I found a "minimum contrast" setting) but for MacOS Terminal there aren't really any good options, short of redefining ANSI "white" to something that isn't white, or switching to a darker background color that is also visibly distinct from any other ANSI color. (Many of us who prefer white-background terminals do so because it matches the brightness level of all the other light-color-themed windows on our display, of which there are many on MacOS.)

Imagine if a command-line tool that you use had been designed (by white-background-terminal-using developers) to output text in ANSI black?

I propose that `styleFile` be redefined from `bold . white` to the following:
```
styleFile :: AnsiDoc -> AnsiDoc
styleFile = bold
```
This should provide the desired visible emphasis no matter what terminal color scheme is in use.

**Comment by mgsloan** [comment]
Oh, yes, I see.  Yes, it makes sense to change that.  One possible solution might be to set the background color dark for files.  Alternatively, picking another color.  I quite like how it currently looks, though.

**Comment by brianhuffman** [comment]
I would be perfectly happy if the white text was also shown with a specified background color (e.g. black).

**Comment by mgsloan** [comment]
@brianhuffman I believe that this PR will resolve the issue https://github.com/commercialhaskell/stack/pull/4079

**Comment by snoyberg** [comment]
Closing, since this appears to have been addressed by #4079.
